x-pack/legacy/plugins/ml/public/data_frame_analytics/common/analytics.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useEffect } from 'react';
+import { BehaviorSubject } from 'rxjs';
+import { filter, distinctUntilChanged } from 'rxjs/operators';
+import { Subscription } from 'rxjs';
+
+// @ts-ignore
+import { isJobIdValid } from '../../../common/util/job_utils';
+
+// TODO
+export const moveToAnalyticsWizard = () => {};
+
+export const isAnalyticsIdValid = isJobIdValid;
+
+export type IndexName = string;
+export type IndexPattern = string;
+export type DataFrameAnalyticsId = string;
+
+export interface CreateRequestBody {
+  description?: string;
+  dest: {
+    index: IndexName;
+    results_field: string;
+  };
+}
+
+export interface DataFrameAnalyticsOutlierConfig extends CreateRequestBody {
+  id: DataFrameAnalyticsId;
+  analysis: {
+    outlier_detection: {};
+  };
+  analyzed_fields: {
+    includes: string[];
+    excludes: string[];
+  };
+  model_memory_limit: string;
+  create_time: number;
+  version: string;
+}
+
+export enum REFRESH_ANALYTICS_LIST_STATE {
+  ERROR = 'error',
+  IDLE = 'idle',
+  LOADING = 'loading',
+  REFRESH = 'refresh',
+}
+export const refreshAnalyticsList$ = new BehaviorSubject<REFRESH_ANALYTICS_LIST_STATE>(
+  REFRESH_ANALYTICS_LIST_STATE.IDLE
+);
+
+export const useRefreshAnalyticsList = (
```

Comment:
this is a nicely structured hook. i wonder if it could be used in a more general way for managing things that load.
also, could `isLoading` be moved to be something returned from the hook, like `refresh` is? i guess you'd then need something watching `isLoading` on the other end.

Reply:
thanks! Yes I think `isLoading` could be moved inside the custom hook, don't remember exactly why I did it this way, maybe I had issues with updates. Just check because someone mentioned it a while ago, APM has a generic version `useFetcher()` similar to this, we could have a look if we wanted to reuse it https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/apm/public/hooks/useFetcher.tsx